src/mainwindow.cpp

```diff
@@ -39,15 +39,84 @@ void MainWindow::emptyTrash()
   trash.clear();
 }
 
+#if defined(HARDWARE_TOUCH)
+void MainWindow::setTouchEnabled(bool enable)
+{
+  switch(touchEnableState)
+  {
+  case TouchOn:
+    if(!enable)
+      touchEnableState = TouchOff;
+    break;
+  case TouchOn_Touching:
+    if(!enable)
+      touchEnableState = TouchOnToOff_Touching;
+    break;
+  case TouchOnToOff_Touching:
+    if(enable)
+      touchEnableState = TouchOn_Touching;
+    break;
+  case TouchOff:
+    if(enable)
+      touchEnableState = TouchOn;
+    break;
+  case TouchOff_Touching:
+    if(enable)
+      touchEnableState = TouchOffToOn_Touching;
+    break;
+  case TouchOffToOn_Touching:
+    if(!enable)
+      touchEnableState = TouchOff_Touching;
+    break;
+  }
+}
+#endif
+
 void MainWindow::checkEvents()
 {
 #if defined(HARDWARE_TOUCH)
+
   if (touchPanelEventOccured()) {
-    touchPanelRead();
+    touchState = touchPanelRead();
```

Comment:
touchPanelRead is void function, so this doesn't work ;)

Reply:
It is not anymore, with the changes from the PR #1000 on EdgeTX. So they depend on each other.

Reply:
Thanks, sorry about that!